pkg/checks/drupal/drupal.go

```diff
@@ -42,37 +43,74 @@ var CheckModulesInYaml = func(c *yaml.YamlBase, ct config.CheckType, configName
 		}
 	}
 
-	for _, m := range required {
-		kvr, _, err := c.CheckKeyValue(moduleKey(m), configName)
-		// It could be a value different from 0, which still means it's enabled.
-		if kvr == yaml.KeyValueEqual || kvr == yaml.KeyValueNotEqual {
-			c.AddPass(fmt.Sprintf("'%s' is enabled", m))
-		} else if kvr == yaml.KeyValueError {
-			c.AddFail(err.Error())
-			c.AddBreach(result.ValueBreach{Value: err.Error()})
-		} else {
-			c.AddFail(fmt.Sprintf("'%s' is not enabled", m))
-			c.AddBreach(result.KeyValueBreach{
-				Key:   "required module is disabled",
-				Value: m,
-			})
+	determineModuleStatus := func(modules []string) ([]string, []string, []string) {
```

Comment:
Should we use a local function instead of a function variable for this so we can test the function discretely?

Reply:
sounds like a good idea, will do that